contribute/style-guides/terminology.md

```diff
@@ -0,0 +1,15 @@
+# Terminology used throughout Grafana code base
```

Comment:
Can we call this "Grafana technical glossary" and move it up a directory?

Reply:
@osg-grafana OK, if that makes sense. I'm still unsure personally.

Reply:
It is not a style guide.

Reply:
Yeah, but I'm unsure where it fits in still.

Reply:
I just tried to look for it at https://grafana.com/docs/grafana/latest/style-guide/.

Reply:
From a content-type perspective, I would expect it at https://github.com/grafana/grafana/tree/master/contribute/technical-glossary.md and in "production" at https://grafana.com/docs/grafana/latest/technical-glossary/ and aliased to https://grafana.com/docs/grafana/latest/glossary/.